Tests/Kernel/crash.cpp

```diff
@@ -362,5 +368,26 @@ int main(int argc, char** argv)
         }).run(run_type);
     }
 
+#if !ARCH(X86_64)
+    if (do_misaligned_fetch) {
+        any_failures |= !Crash("Perform a misaligned instruction fetch  ", []() {
+#    if ARCH(AARCH64)
+            TODO_AARCH64();
+#    elif ARCH(RISCV64)
+            // An offset of 8 should simply jump to the next instruction.
+            // 9 is misaligned, but will have the same effect if the test fails.
+            // FIXME: This relies on addi and jr being compiled to compressed instructions. Figure out a better way to do this.
+            asm volatile(
+                "auipc t0, 0\n"
+                "addi t0, t0, 9\n"
+                "jr t0");
```

Comment:
If you require those to be compressed, directly using `c.addi` and `c.jr` may be nicer.

Reply:
oops i tried suffixing them with .c, you can tell how little experience i have with C

Reply:
Alternatively, you could rewrite it to use uncompressed instructions.
You can disable the generation of compressed instructions with [`.option norvc`](https://github.com/riscv-non-isa/riscv-asm-manual/blob/main/riscv-asm.md#rvcnorvc).

Reply:
Also, I think this will never crash with the C extension.
IIRC with IALIGN=16, misaligned jumps are not allowed to trap.

Reply:
Also, I think this will never crash with the C extension.
IIRC with IALIGN=16, misaligned jumps are not allowed to trap.